dao/src/main/java/org/wso2/testgrid/dao/uow/TestPlanUOW.java

```diff
@@ -128,7 +128,15 @@ public Status getCurrentStatus(Product product) {
         List<TestPlan> succesfulPlans = testPlans.stream().filter(testPlan ->
                 testPlan.getStatus().equals(Status.SUCCESS)
         ).collect(Collectors.toList());
-        return succesfulPlans.size() == testPlans.size() ? Status.SUCCESS : Status.FAIL;
+        //check if there are pending test_plans
+        List<TestPlan> pendingPlans = testPlans.stream().filter(testPlan ->
+                testPlan.getStatus().equals(Status.RUNNING)
+        ).collect(Collectors.toList());
```

Comment:
Instead of collecting, we can use Lambda #[anyMatch](https://docs.oracle.com/javase/8/docs/api/java/util/stream/Stream.html#anyMatch-java.util.function.Predicate-) instead.